File: competitions/CCP 2021/kotakdalamkotak.py
import math
n = int(input())    
i = 1
least = 1e9
if math.sqrt(n)**2 == n:
    a = math.sqrt(n)
    b = math.sqrt(n)
else:
    while i <= math.sqrt(n):
        if (n % i == 0) :
            if (n // i == i) :
                a = i
                b = n//i
                # print(i)
                # list.append(i)
                # list.append(i//n)
            else:
                # print(i , n//i)
                # list.append(i)
                # list.append(n//i)
                if min(abs(i-n//i), least) < least:
                    a = i
                    b = n//i
                    least =  min(abs(i-n//i), least)
        i = i + 1
print(int(a), int(b))
# for i in range(len(list)):
#     if min(abs(list[i+1]-list[i]), least):
#         a = i 
#         b = i+1
#         least = min(abs(list[i]-list[i+1]))

# print(a, b)
    

File: competitions/CCP 2021/kotakDalamKotak.cpp
#include <bits/stdc++.h>
#define ll long long
#define ld long double
const int MOD = 1e9;
using namespace std;

void solve(){
    ll luas; cin >> luas;
    ll p, l;
    ll least = MOD;
    // p = luas/l
    // l = luas/p
    if (sqrt(luas) != 0 && pow(sqrt(luas),2) == luas) {
        p = sqrt(luas);
        l = sqrt(luas);
    }  else {
        for (int i = 1; i < luas/2; i++) {
            if (luas / i == i) {
                p = i;
                l = luas/i;
            } else {
                if (min(abs(i-luas/i), least) < least) {
                    p = i;
                    l = luas / i;
                    least = min(abs(i-(luas/i)), least);
                }
            }
        }
    }
    //     p = luas/i;
    //     l = luas/p;
    //     if (luas == (p * l)) {
    cout << p << " " << l << endl;
    //     }
}

int main(){
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    solve();

    return 0;
}   